### PR TITLE
Remove GAed feature gates SeccompDefault/TopologyManager/OpenAPIV3/Se…

### PR DESCRIPTION
**Pull request by tukwila**
…rverSideApply/ServerSideFieldValidation

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

All of these feature gates have been GAed, and can be removed since v1.29:

- SeccompDefault
- TopologyManager
- OpenAPIV3
- ServerSideApply
- ServerSideFieldValidation

#### Which issue(s) this PR fixes:

Fixes # None

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by tukwila** [comment]
@saschagrunert could you please look at this PR that i plan to remove SeccompDefault from GA? 
@lmdaly @swatisehgal please look at this PR as well?

**Review by saschagrunert — APPROVED** [review]

File: test/integration/apiserver/openapi/openapi_merge_test.go [hunk 1]
```diff
@@ -1,199 +0,0 @@
...
-	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
-	"k8s.io/kubernetes/test/integration/framework"
-)
-
-const scaleSchemaName = "io.k8s.api.autoscaling.v1.Scale"
-const statusSchemaName = "io.k8s.apimachinery.pkg.apis.meta.v1.Status"
-const objectMetaSchemaName = "io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta"
-
-func TestOpenAPIV3MultipleCRDsSameGV(t *testing.T) {
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.OpenAPIV3, true)()
```
Comment (liggitt) [inline test/integration/apiserver/openapi/openapi_merge_test.go:45]: don't drop this test, it was testing with this feature enabled and should remain, just drop the SetFeatureGateDuringTest call

File: test/integration/apiserver/openapi/openapiv3_test.go [hunk 2]
```diff
@@ -1,268 +0,0 @@
...
-	"k8s.io/kubernetes/test/utils/ktesting"
-	"sigs.k8s.io/yaml"
-)
-
-func TestOpenAPIV3SpecRoundTrip(t *testing.T) {
-	_, ctx := ktesting.NewTestContext(t)
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.OpenAPIV3, true)()
```
Comment (liggitt) [inline test/integration/apiserver/openapi/openapiv3_test.go:53]: don't drop this test, it was testing with this feature enabled and should remain, just drop the SetFeatureGateDuringTest call

**Review by liggitt — COMMENTED** [review]
a couple comments on the test removals

in general, it's better to have one PR drop one feature gate (or related set of feature gates), since those relate to different SIGs

@logicalhan @jpbetz, what are your thoughts about continuing to drop feature gates on schedule, knowing there's a desire to track them past GA in the future? The code these gate was already ungated when they got promoted to GA, so I don't think we're losing much here

File: test/integration/apiserver/openapi/openapi_merge_test.go (hunk 1 above)
Comment (tukwila) [reply test/integration/apiserver/openapi/openapi_merge_test.go:45]: yes, recovered.

File: test/integration/apiserver/openapi/openapiv3_test.go (hunk 2 above)
Comment (tukwila) [reply test/integration/apiserver/openapi/openapiv3_test.go:53]: yes, recovered.

File: staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go [hunk 3]
```diff
@@ -141,12 +140,6 @@ func (o *WardleServerOptions) Config() (*apiserver.Config, error) {
 	serverConfig.OpenAPIConfig.Info.Title = "Wardle"
 	serverConfig.OpenAPIConfig.Info.Version = "0.1"
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.OpenAPIV3) {
```
Comment (pacoxu) [inline staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go:144]: Why remove the below logic?

OpenAPIV3 is always enabled here.

Comment (tukwila) [reply staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go:144]: corrected.

**Comment by pacoxu** [comment]
Overall LGTM.

> @logicalhan @jpbetz, what are your thoughts about continuing to drop feature gates on schedule, knowing there's a desire to track them past GA in the future? The code these gate was already ungated when they got promoted to GA, so I don't think we're losing much here

Pending on feature owners and Jordon's comment.

**Pull request closed without merging**